[Bug] Crash: Can't create filtered deck

###### Reproduction Steps

1. Create Filtered Deck
2. Press OK
3. 


###### Expected Result

No Crash

###### Actual Result

Crash

###### Debug info
AnkiDroid Version = 2.16alpha53

Android Version = 11

Manufacturer = Google

Model = Pixel 3a

Hardware = sargo

Webview User Agent = Mozilla/5.0 (Linux; Android 11; Pixel 3a Build/RQ3A.211001.001; wv) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/99.0.4844.88 Mobile Safari/537.36

ACRA UUID = 3e7c4ead-530d-4748-9f4f-622f9f619871

Scheduler = std2

Crash Reports Enabled = false

DatabaseV2 Enabled = true

###### Research
*Enter an [x] character to confirm the points below:*

- [x] I have read the [support page](https://ankidroid.org/docs/help.html) and am reporting a bug or enhancement request specific to AnkiDroid
- [x] I have checked the [manual](https://ankidroid.org/docs/manual.html) and the [FAQ](https://github.com/ankidroid/Anki-Android/wiki/FAQ) and could not find a solution to my issue
- [x] I have searched for similar existing issues here and on the user forum
- [x] (Optional) I have confirmed the issue is not resolved in the latest alpha release ([instructions](https://docs.ankidroid.org/manual.html#betaTesting))

```
2022-03-30 18:49:03.850 21065-21065/com.ichi2.anki E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.ichi2.anki, PID: 21065
    java.lang.RuntimeException: Unable to start activity ComponentInfo{com.ichi2.anki/com.ichi2.anki.FilteredDeckOptions}: java.lang.NullPointerException: Parameter specified as non-null is null: method kotlin.jvm.internal.Intrinsics.checkNotNullParameter, parameter key
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3431)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3595)
        at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:85)
        at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2066)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loop(Looper.java:223)
        at android.app.ActivityThread.main(ActivityThread.java:7664)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:947)
     Caused by: java.lang.NullPointerException: Parameter specified as non-null is null: method kotlin.jvm.internal.Intrinsics.checkNotNullParameter, parameter key
        at com.ichi2.anki.FilteredDeckOptions.onSharedPreferenceChanged(Unknown Source:11)
        at com.ichi2.anki.FilteredDeckOptions$DeckPreferenceHack$Editor.commit(FilteredDeckOptions.kt:173)
        at com.ichi2.anki.FilteredDeckOptions$DeckPreferenceHack$Editor.apply(FilteredDeckOptions.kt:211)
        at android.preference.Preference.tryCommit(Preference.java:1621)
        at android.preference.Preference.persistString(Preference.java:1657)
        at android.preference.ListPreference.setValue(ListPreference.java:157)
        at com.ichi2.anki.FilteredDeckOptions.buildLists(FilteredDeckOptions.kt:443)
        at com.ichi2.anki.FilteredDeckOptions.onCreate(FilteredDeckOptions.kt:307)
        at android.app.Activity.performCreate(Activity.java:7994)
        at android.app.Activity.performCreate(Activity.java:7978)
        at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1309)
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3404)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3595) 
        at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:85) 
        at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135) 
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2066) 
        at android.os.Handler.dispatchMessage(Handler.java:106) 
        at android.os.Looper.loop(Looper.java:223) 
        at android.app.ActivityThread.main(ActivityThread.java:7664) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:947) 
```

--- a/AnkiDroid/src/main/java/com/ichi2/anki/FilteredDeckOptions.kt
+++ b/AnkiDroid/src/main/java/com/ichi2/anki/FilteredDeckOptions.kt
@@ -27,6 +27,7 @@ import com.ichi2.anim.ActivityTransitionAnimation
 import com.ichi2.anim.ActivityTransitionAnimation.slide
 import com.ichi2.anki.analytics.UsageAnalytics
 import com.ichi2.anki.receiver.SdCardReceiver
+import com.ichi2.annotations.NeedsTest
 import com.ichi2.libanki.Collection
 import com.ichi2.libanki.Deck
 import com.ichi2.preferences.StepsPreference.Companion.convertFromJSON
@@ -40,6 +41,7 @@ import com.ichi2.utils.KotlinCleanup
 import timber.log.Timber
 import java.util.*
 
+@NeedsTest("construction + onCreate - do this after converting to fragment-based preferences.")
 class FilteredDeckOptions : AppCompatPreferenceActivity(), OnSharedPreferenceChangeListener {
     @KotlinCleanup("try to make mDeck non-null / use lateinit")
     private var mDeck: Deck? = null

--- a/AnkiDroid/src/main/java/com/ichi2/anki/FilteredDeckOptions.kt
+++ b/AnkiDroid/src/main/java/com/ichi2/anki/FilteredDeckOptions.kt
@@ -359,7 +359,8 @@ class FilteredDeckOptions : AppCompatPreferenceActivity(), OnSharedPreferenceCha
         return false
     }
 
-    override fun onSharedPreferenceChanged(sharedPreferences: SharedPreferences, key: String) {
+    @KotlinCleanup("Find a different method rather than providing a null key in a caller")
+    override fun onSharedPreferenceChanged(sharedPreferences: SharedPreferences, key: String?) {
         // update values on changed preference
         updateSummaries()
         mPrefChanged = true

--- a/AnkiDroid/src/main/java/com/ichi2/annotations/NeedsTest.kt
+++ b/AnkiDroid/src/main/java/com/ichi2/annotations/NeedsTest.kt
@@ -1,0 +1,32 @@
+/*
+ *  Copyright (c) 2022 David Allison <davidallisongithub@gmail.com>
+ *
+ *  This program is free software; you can redistribute it and/or modify it under
+ *  the terms of the GNU General Public License as published by the Free Software
+ *  Foundation; either version 3 of the License, or (at your option) any later
+ *  version.
+ *
+ *  This program is distributed in the hope that it will be useful, but WITHOUT ANY
+ *  WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+ *  PARTICULAR PURPOSE. See the GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License along with
+ *  this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package com.ichi2.annotations
+
+/**
+ * Use when code needs unit tests
+ *
+ * To ensure the reminder isn't forgotten about or left to go stale in a GitHub issue
+ */
+@Target(
+    AnnotationTarget.CLASS, AnnotationTarget.FUNCTION,
+    AnnotationTarget.VALUE_PARAMETER, AnnotationTarget.EXPRESSION,
+    AnnotationTarget.FIELD, AnnotationTarget.PROPERTY
+)
+@Repeatable
+@Retention(AnnotationRetention.SOURCE)
+@MustBeDocumented
+annotation class NeedsTest(val value: String)

